### PR TITLE
Site Editor: don't hide nav after Gutenberg 12.2

### PR DESCRIPTION
**Pull request by mattwiebe**
#### Changes proposed in this Pull Request

* Unhide the Navigation menu (under the top left W)

Gutenberg 12.2 (ahead of and for WP 5.9) has a new, drastically scaled-back Navigation Sidebar. The `< Dashboard` link is no longer nested-away-by-default, so we can stop hijacking the (W) menu to go back to the Dashboard directly.

This also prevents the weird state in which you could open the nav sidebar by picking "browse templates" from the top middle template dropdown without being able to close it again. There was probably an issue about that.

#### Testing instructions
1. Ensure your test sandboxed site has the `gutenberg-edge` sticker. 
2. Wait until WordPress/gutenberg/pull/36488 is merged and released (in a RC2?), or build it yourself and extract the built artifact from `npm run build:plugin-zip` in the `gutenberg-core/v12.1.0-rc.1` directory on your sandbox to overwrite it. (If you sync a working repo without building, you won't have the required `GUTENBERG_VERSION` PHP constant.)
3. This ETK build is running on your sandbox
4. Once that patched Gutenberg is running, and the ETK is patched, et voila. The Navigation Sidebar is now working just like Core and the `< Dashboard` link takes you back to Calypso home.

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php
Comment (zaguiini) [inline apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: Is this comment necessary? I feel that people can always refer to the `git blame`, check the PR and the explanation on why it is not needed anymore. Comments are often misleading, get outdated pretty quickly, and pollute the source code.

**Review by zaguiini — APPROVED** [review]
Works perfectly on my end 👍

**Review by vindl — APPROVED** [review]

File: apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php
Comment (vindl) [inline apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: This should be updated to `12.2.0`. I also think that instead of conditional loading we can delete this code and folder after it lands 12.2 lands in production.

File: apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php
Comment (mattwiebe) [reply apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: The point of this code comment is to encourage future maintainers to investigate if this can be removed, should I not get to it. But your point is a fair one, given the number of years-later `@todo` and `@fixme` comments strewn about.

Comment (zaguiini) [reply apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: How about an issue instead of a comment? You could even link it as a comment inside the code.

File: apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php
Comment (mattwiebe) [reply apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: Yup this should all be removed after `12.2.0` hits production. This is simply a mechanism to deploy for `edge` and `production` simultaneously.

File: apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php
Comment (Addison-Stavlo) [reply apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: I think its a handy companion to the version check below. As long as we are conditionally enabling functionality based off a magic gutenberg version number, it seems good to note why for easy reference.

Hopefully it wont get outdated or misleading, as it is accurate to the code beneath it. When that code is updated, then the comment should be as well. Ideally, once we don't support any pre-12.2 Gutenberg version on dotcom anymore we can just remove this entire site editor section of this plugin.

Comment (Addison-Stavlo) [reply apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: > How about an issue instead of a comment? 

An issue sounds good to me in addition, so we can just get rid of all of this in the near future - https://github.com/Automattic/wp-calypso/issues/59295

Comment (mattwiebe) [reply apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php]: Thanks for creating that issue @Addison-Stavlo ! I intended to do the same but this has been the week of inability to maintain an attention span.

**Pull request merged**
